internal/controllers/liveness/namespace.go

```diff
@@ -84,42 +101,47 @@ func (c *namespaceController) Reconcile(ctx context.Context, req ctrl.Request) (
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("creating namespace: %w", err)
 	}
-	logger.V(0).Info("recreating missing namespace to free orphaned symphony")
+	logger.V(0).Info("recreated missing namespace to free orphaned resources")
 	return ctrl.Result{}, nil
 }
 
 const removeFinalizersPatch = `[{ "op": "remove", "path": "/metadata/finalizers" }]`
 
 func (c *namespaceController) cleanup(ctx context.Context, ns string) error {
-	logger := logr.FromContextOrDiscard(ctx).WithValues("symphonyNamespace", ns)
-	logger.V(0).Info("deleting any remaining symphonies in orphaned namespace")
-	err := c.client.DeleteAllOf(ctx, &apiv1.Symphony{}, client.InNamespace(ns))
-	if err != nil {
-		return fmt.Errorf("deleting symphonies: %w", err)
-	}
-
-	list := &apiv1.ResourceSliceList{}
-	err = c.client.List(ctx, list, client.InNamespace(ns))
-	if err != nil {
-		return fmt.Errorf("listing resource slices: %w", err)
-	}
+	logger := logr.FromContextOrDiscard(ctx).WithValues("resourceNamespace", ns)
 
-	for _, item := range list.Items {
-		if len(item.Finalizers) == 0 {
-			continue
-		}
-		err = c.client.Patch(ctx, &item, client.RawPatch(types.JSONPatchType, []byte(removeFinalizersPatch)))
+	logger.V(1).Info("deleting any remaining resources in orphaned namespace")
+	for _, kind := range orphanableKinds {
+		err := c.client.DeleteAllOf(ctx, &metav1.PartialObjectMetadata{
+			TypeMeta: metav1.TypeMeta{Kind: kind, APIVersion: apiv1.SchemeGroupVersion.String()},
+		}, client.InNamespace(ns))
 		if err != nil {
-			return fmt.Errorf("removing finalizers from resource slice %q: %w", item.Name, err)
+			return fmt.Errorf("deleting resources of kind %q: %w", kind, err)
 		}
-		logger := logger.WithValues("resourceSliceName", item.Name)
-		logger.V(0).Info("forcibly removed finalizers")
 	}
-
 	return nil
 }
 
-func mostRecentCreation(list *apiv1.SymphonyList) time.Time {
+func (c *namespaceController) findOrphans(ctx context.Context, ns, kind string) (bool, *ctrl.Result, error) {
+	logger := logr.FromContextOrDiscard(ctx).WithValues("namespace", ns)
+	list := &metav1.PartialObjectMetadataList{}
+	list.Kind = kind
+	list.APIVersion = "eno.azure.io/v1"
+	err := c.client.List(ctx, list, client.InNamespace(ns))
+	if err != nil {
+		return false, nil, fmt.Errorf("listing resources of kind %q: %w", kind, err)
+	}
+	if len(list.Items) == 0 {
+		return false, nil, nil // no orphaned resources, nothing to do
+	}
+	if delta := time.Since(mostRecentCreation(list)); delta < time.Second {
```

Comment:
Should we make the delta threshold adjustable?

Reply:
lol That's completely reasonable - will do

Reply:
Is 1 sec an informed guess? I'm unsure if etcd streaming for different Kinds could drift this much under different levels of load.